impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -0,0 +1,283 @@
+                                                   {Point(1, 1), Point(1, 1)},
+                                               });
+
+        GaussianBlurFragmentShader::BindTextureSampler(
+            cmd, input_texture,
+            renderer.GetContext()->GetSamplerLibrary()->GetSampler(
+                sampler_descriptor));
+        GaussianBlurVertexShader::BindFrameInfo(
+            cmd, host_buffer.EmplaceUniform(frame_info));
+        GaussianBlurFragmentShader::BindBlurInfo(
+            cmd, host_buffer.EmplaceUniform(blur_info));
+        pass.AddCommand(std::move(cmd));
+
+        return true;
+      };
+  std::shared_ptr<Texture> out_texture = renderer.MakeSubpass(
+      "Gaussian Blur Filter", subpass_size, subpass_callback);
+  return out_texture;
+}
+
+/// Given a desired |scalar|, will return the scalar that gets close but leaves
+/// |size| in integer sizes.
+Vector2 CalculateIntegerScale(Scalar scalar, ISize size) {
+  ISize new_size(size.width / scalar, size.height / scalar);
+  return Vector2(size.width / static_cast<Scalar>(new_size.width),
+                 size.height / static_cast<Scalar>(new_size.height));
+}
+
+/// Calculate how much to scale down the texture depending on the blur radius.
+/// This curve was taken from |DirectionalGaussianBlurFilterContents|.
+Scalar CalculateScale(Scalar radius) {
+  constexpr Scalar decay = 4.0;   // Larger is more gradual.
+  constexpr Scalar limit = 0.95;  // The maximum percentage of the scaledown.
+  const Scalar curve =
+      std::min(1.0, decay / (std::max(1.0f, radius) + decay - 1.0));
+  return (curve - 1) * limit + 1;
+};
+
+}  // namespace
+
+GaussianBlurFilterContents::GaussianBlurFilterContents(Scalar sigma)
+    : sigma_(sigma) {}
+
+std::optional<Rect> GaussianBlurFilterContents::GetFilterSourceCoverage(
+    const Matrix& effect_transform,
+    const Rect& output_limit) const {
+  Scalar blur_radius = CalculateBlurRadius(sigma_);
+  Vector3 blur_radii =
+      effect_transform.Basis() * Vector3{blur_radius, blur_radius, 0.0};
+  return output_limit.Expand(Point(blur_radii.x, blur_radii.y));
+}
+
+std::optional<Rect> GaussianBlurFilterContents::GetFilterCoverage(
+    const FilterInput::Vector& inputs,
+    const Entity& entity,
+    const Matrix& effect_transform) const {
+  if (inputs.empty()) {
+    return {};
+  }
+
+  std::optional<Rect> input_coverage = inputs[0]->GetCoverage(entity);
+  if (!input_coverage.has_value()) {
+    return {};
+  }
+
+  Scalar blur_radius = CalculateBlurRadius(sigma_);
+  Vector3 blur_radii =
+      effect_transform.Basis() * Vector3{blur_radius, blur_radius, 0.0};
+  return input_coverage.value().Expand(Point(blur_radii.x, blur_radii.y));
+}
+
+std::optional<Entity> GaussianBlurFilterContents::RenderFilter(
+    const FilterInput::Vector& inputs,
+    const ContentContext& renderer,
+    const Entity& entity,
+    const Matrix& effect_transform,
+    const Rect& coverage,
+    const std::optional<Rect>& coverage_hint) const {
```

Comment:
I don't see the snapshot's transform being consumed anywhere, which will cause issues with some optimized snapshots.

The "destination rect" of `TextureContents` scales the texture when taking a snapshot to avoid an unnecessary render pass. I'd recommend absorbing the scale of the basis vectors in the first downsampling pass, and applying other aspects of the transform to the final returned entity.

So the texture of a snapshot may be in pass space, entity space, or something in-between.

![image](https://github.com/flutter/engine/assets/919017/f7dd449e-d5f0-4937-a057-63514ee3ed08)

Reply:
What parameter are you editing to get that?  I did see that but I didn't have a case where it manifested.  I assumed that was getting absorbed elsewhere so the uv coords weren't effected.

Reply:
It's the "Path XYWH" at the bottom